feat: add support for discarding flags while inheriting (#14465)
WeblateOrg/weblate: docs/admin/checks.rst

To ensure that internal links are not being translated (i.e. `[test](../checks)`
does not become `[test](../chequeos)`.

> The flags defined in the higher scope can be discarded using the
> ``discard:NAME`` syntax. For example if component is configured to
> ``safe-html``, you can add ``discard:safe-html`` to the string flags to skip it
> for this particular string.

Reviewer: ```suggestion
The flags defined on a higher level can be discarded using the
``discard:NAME`` syntax. For example, if a component is configured to
``safe-html``, you can add ``discard:safe-html`` to the string flags to skip it
for this particular string.
```

    Mark this string as a variant of string with matching source. See :ref:`variants`.
``regex:REGEX``
    Regular expression to match translation, see :ref:`check-regex`.
``discard:NAME``

>     Discards flag defined in the higher scope.

Reviewer: ```suggestion
    Discards flag defined on a higher level.
```
